src/Middleware/ResponseCompression/src/BodyWrapperStream.cs

```diff
@@ -326,13 +338,14 @@ private async Task InnerSendFileAsync(string path, long offset, long? count, Can
         public Task StartAsync(CancellationToken token = default)
         {
             OnWrite();
+            return _innerBodyFeature.StartAsync(token);
+        }
 
-            if (_innerStartFeature != null)
-            {
-                return _innerStartFeature.StartAsync(token);
-            }
-
-            return Task.CompletedTask;
+        public async Task CompleteAsync()
+        {
+            await StartAsync();
```

Comment:
Hm. Maybe we should add HasStarted/HasCompleted bools to the interface?

Reply:
Or do that internally.

Reply:
Internally. We already have IHttpResponseFeature.HasStarted which means something different (headers have been locked).

Reply:
StartAsync locks headers too, correct? I thought StartAsync and HasStarted were equivalent.

Reply:
Only if StartAsync ends up cascading all the way to the server.